include/pybind11/attr.h

```diff
@@ -40,7 +40,7 @@ template <typename T> struct arg_t : public arg {
 
 inline namespace literals {
 /// String literal version of arg
-constexpr arg operator"" _a(const char *name, size_t) { return {name}; }
+constexpr arg operator"" _a(const char *name, size_t) { return {arg(name)}; }
```

Comment:
Minor nitpick: you can omit the braces here so that it's just `return arg(name);`. Strictly speaking, the braces would create another copy since `{arg(name)}` is equivalent to `arg(arg(name))`.